lib/puppet/provider/iis_site/webadministration.rb

```diff
@@ -38,6 +38,59 @@ def create
     return exists?
   end
 
+  def update
+    Puppet.err 'UPDATE FUNCTION'
+
+    # Instantiate server manager object and get matching site using resource name
+    inst_cmd = []
+    inst_cmd << '[Void][Reflection.Assembly]::LoadWithPartialName("Microsoft.Web.Administration");'
+    inst_cmd << '$serverManager = New-Object Microsoft.Web.Administration.ServerManager;'
+    inst_cmd << "$site = $serverManager.Sites | where { $_.Name -eq '#{@resource[:name]}' };"
+    inst_cmd << "$app = $site.Applications | where { $_.Path -eq '/' };"
+    inst_cmd << "$vdir = $app.VirtualDirectories | where { $_.Path -eq '/' };"
+
+    # Set sites properties and commit them
+    @resource.properties.select { |rp| rp.name != :ensure }.each do |property|
+      case property.name
+        when :physicalpath
+          inst_cmd << "$vdir.PhysicalPath='#{property.value}';"
+        when :applicationpool
+          inst_cmd << "$app.ApplicationPoolName='#{property.value}';"
+        when :serverautostart
+          inst_cmd << "$app.EnabledProtocols='#{property.value}';"
+        when :enabledprotocols
+          inst_cmd << "$app.ServerAutoStart='#{property.value}';"
+        when :logpath
+          inst_cmd << "$site.LogFile.Directory='#{property.value}';"
+        when :logperiod
+          inst_cmd << "$site.LogFile.Period='#{property.value}';"
+        when :logtruncatesize
+          inst_cmd << "$site.LogFile.TruncateSize='#{property.value}';"
+        when :loglocaltimerollover
+          inst_cmd << "$site.LogFile.LocalTimeRollover='#{property.value}';"
+        when :logformat
+          inst_cmd << "$site.LogFile.LogFormat='#{property.value}';"
+        when :logflags
+          inst_cmd << "$site.LogFile.LogExtFileFlags='#{property.value}';"
+
+        else
+          inst_cmd << "$site.#{property.name.to_s}='#{property.value}';"
+      end
+    end
+    inst_cmd << '$serverManager.CommitChanges();'
+    inst_cmd = inst_cmd.join
+
+    Puppet.err('COMMAND: ' + inst_cmd)
+    result = self.class.run(inst_cmd)
+    Puppet.err('RESULT: ' + result[:stdout])
+
+    # Handle errors - had issue where when an issue occurred, errors were only output to stdout, not stderr
```

Comment:
This is powershell, not bash. We have to handle errors. See comment about erb above